src/sys/windows/ll.rs

```diff
@@ -0,0 +1,219 @@
+        let entry = Entry::new(self, &mut node);
+
+        match unsafe { self.tail.resolve_mut() } {
+            None => self.push_front(node),
+            Some(tail) => {
+                tail.set_next(node);
+                self.tail = Rawlink::from(&mut tail.next);
+            }
+        }
+
+        entry
+    }
+
+    pub fn remove(&mut self, mut entry: Entry<T>) -> T {
+        self.ensure_same_ll(&entry);
+
+        let (mut prev, next) = unsafe {
+            let node = entry.node.resolve_mut().expect("invalid entry");
+            (node.prev, node.next.take())
+        };
+
+        // Unlink previous pointer
+        let removed = match unsafe { (prev.resolve_mut(), next) } {
+            (Some(p), Some(mut next)) => {
+                next.prev = prev;
+                *mem::replace(&mut p.next, Some(next)).take().unwrap()
+            }
+            (None, Some(mut next)) => {
+                next.prev = Rawlink::none();
+                *mem::replace(&mut self.head, Some(next)).take().unwrap()
+            }
+            (Some(p), None) => {
+                self.tail = prev;
+                *p.next.take().unwrap()
+            }
+            (None, None) => {
+                self.tail = Rawlink::none();
+                *self.head.take().unwrap()
+            }
+        };
+
+        debug_assert!(removed.next.is_none());
+        removed.value
+    }
+
+    pub fn iter(&self) -> Iter<T> {
+        Iter { curr: &self.head }
+    }
+
+    fn push_front(&mut self, mut new_head: Box<Node<T>>) {
+        match self.head {
+            None => {
+                self.head = link_no_prev(new_head);
+                self.tail = Rawlink::from(&mut self.head);
+            }
+            Some(ref mut head) => {
+                new_head.prev = Rawlink::none();
+                head.prev = Rawlink::some(&mut *new_head);
+                mem::swap(head, &mut new_head);
+                head.next = Some(new_head);
+            }
+        }
+    }
+
+    fn ensure_same_ll(&self, entry: &Entry<T>) {
+        assert!(entry.ll == self as *const LinkedList<T> as *mut LinkedList<T>, "entry belongs to a different LinkedList");
+    }
+}
+
+unsafe impl<T> Send for LinkedList<T> {}
+
+pub struct Entry<T> {
+    ll: *mut LinkedList<T>,
+    node: Rawlink<Node<T>>,
+}
+
+impl<T> Entry<T> {
+    fn new(ll: &mut LinkedList<T>, node: &mut Node<T>) -> Entry<T> {
+        Entry {
+            ll: ll as *mut LinkedList<T>,
+            node: Rawlink::some(node),
+        }
+    }
+}
+
+/// An iterator over references to the items of a `LinkedList`.
+pub struct Iter<'a, T:'a> {
+    curr: &'a Link<T>,
+}
+
+impl<'a, A> Iterator for Iter<'a, A> {
+    type Item = &'a A;
+
+    fn next(&mut self) -> Option<&'a A> {
+        self.curr.as_ref().map(|curr| {
+            self.curr = &curr.next;
+            &curr.value
+        })
+    }
+}
+
+unsafe impl<T> Send for Entry<T> {}
+unsafe impl<T> Sync for Entry<T> {}
+
+type Link<T> = Option<Box<Node<T>>>;
```

Comment:
Man I always find linked lists somewhat difficult to read, we've got `Link`, `Rawlink`, `Node`, `Box`, ...